OneCommunityGlobal/HGNRest
# Navneeth add google doc link to weekly summaries email

# Description
![task description](https://github.com/OneCommunityGlobal/HGNRest/assets/44264834/807b989f-3e0f-41e2-ab2a-29e5b1197584)

## Related PRs (if any):
None. Does not require the Frontend PR to run.

## Main changes explained:
- Fetch the admin links from the weeklySummaries results.
- Retrieve the Google Doc Link from the admin links if available,
- Update the email body to include Google Doc Link in the Weekly Summaries.
- Remove Sangam's email from the recipients list.

## How to test:
1. Check into the current branch
2. Ensure that Nodemailer is set up (see the Notes section) so that you have the functionality to send an email from the app
3. Make the following code changes in `src/cronjobs/userProfileJobs.js`:
- Un-comment line 8 `// '* * * * *', // Comment out for testing. Run Every minute.` 
This is to run the cronjob every minute for convenient testing. Please ensure that you are stopping the execution of the code immediately after an email is sent. It is advisable to set up custom schedule values (see the Notes section) that are based on the time and date when you are testing the PR.
- Comment line 9 `'1 0 * * 0', // Every Sunday, 1 minute past midnight.`
-  Comment line 13 `await userhelper.assignBlueSquareForTimeNotMet();`
- Comment line 14 `await userhelper.applyMissedHourForCoreTeam();`
- Comment line 16 `await userhelper.deActivateUser();`
- Comment line 17 `await userhelper.deleteExpiredTokens();`
- Comment line 19 `await userhelper.awardNewBadges();`
- Comment line 20 `await userhelper.reActivateUser();`
- Comment line 21 `await userhelper.deActivateUser();`
- Set the day of the week in line 11 `const SUNDAY = 0;` as the integer day when you will be testing this PR where counting begins from 0 and the week begins from Sunday (0 for Sunday, 1 for Monday, and so on...)
4. Make the following code changes in `src/helpers/userhelper.js`:
- Comment line 273 `"onecommunityglobal@gmail.com, onecommunityhospitality@gmail.com",`
- Add a line below it with your email in the same format as above: `"youremail@email.com",`    
5. Save the changes in both the files, run `npm install`, and run `npm run dev` to run this PR locally
6. Once you receive the email, verify if the Google Doc Link is a field added to the Weekly Summaries email (see the Notes section for the expected format)
7. MOST IMPORTANT: Immediately stop the execution.

## Test Video (with instructions narrated):

https://github.com/OneCommunityGlobal/HGNRest/assets/44264834/5fa89add-57b2-426c-8628-816ea156cc90

## Notes
### Syntax for setting schedule values in a cronjob
![Cron Job syntax](https://github.com/OneCommunityGlobal/HGNRest/assets/44264834/b97b0b4d-8e40-430c-ba0a-ade5d9036f5f)
- Please follow [this guide](https://www.ibm.com/docs/en/db2oc?topic=task-unix-cron-format) to learn more about the UNIX cron format.

### Google Doc Link in the received email (expected format)
![google doc link in the received email](https://github.com/OneCommunityGlobal/HGNRest/assets/44264834/28c87fb4-ac8b-4ec8-acca-b2384eb3733e)

### Nodemailer setup
You need to have Nodemailer set up to add the following environment variables to the `.env` file in your backend setup in order to send an email:
```
REACT_APP_EMAIL
REACT_APP_EMAIL_CLIENT_ID
REACT_APP_EMAIL_CLIENT_SECRET
REACT_APP_EMAIL_CLIENT_REDIRECT_URI
REACT_APP_EMAIL_REFRESH_TOKEN
```

- Set your sender email ID as the value for the REACT_APP_EMAIL environment variable.
- Please follow this tutorial to set up Nodemailer and OAuth2 to set up and add the Client ID, Client Secret, Redirect URI, and Refresh Token: [Link to YouTube Video](https://youtu.be/-rcRf7yswfM?si=I9LdYTpn60EVVc3a)
- Ensure that the following environment variable with value is added to the `.env` file:
```sendEmail=TRUE```
